src/components/victory-bar/victory-bar.js

```diff
@@ -408,15 +408,17 @@ export default class VictoryBar extends React.Component {
         {}, dataProps, {events: Events.getPartialEvents(dataEvents, key, dataProps)}
       ));
 
-      const labelProps = defaults(
-        {index, key: `${role}-label-${key}`},
-        this.getEventState(key, "labels"),
-        this.getSharedEventState(key, "labels"),
-        labelComponent.props,
-        this.baseProps[key].labels
-      );
+      const shouldGetLabelProps = (this.baseProps[key].labels && this.baseProps[key].labels.text)
+        || this.props.events;
 
-      if (labelProps && labelProps.text) {
```

Comment:
this `if` still needs to exist to avoid creating labelComponents when there is no text, because eg. when there are `events` but they don't create labels